net/davix/src/TDavixFile.cxx

```diff
@@ -736,7 +737,7 @@ void TDavixFile::Seek(Long64_t offset, ERelativeTo pos)
 
 Bool_t TDavixFile::ReadBuffer(char *buf, Int_t len)
 {
-   TLockGuard guard(&(d_ptr->positionLock));
+   TLockGuard guard(&davixLock);
```

Comment:
Isn't this needed also in the next function line 759: `Bool_t TDavixFile::ReadBuffer(char *buf, Long64_t pos, Int_t len)` ?

Reply:
and/or why here and not every single other function :( ?

Reply:
Also the lock scope needs to be restricted.  The line 748/749:
```
Info("ReadBuffer", "%lld bytes of data read sequentially"
```
can take the ROOT global lock and thus can lead to dead lock if it is called within another locks.

Reply:
Thanks a lot for the comments! The last commit is addressing these.

> and/or why here and not every single other function :( ?

Well, I haven't looked into this in all detail. I just tried to add the minimal amount of locks to "never" hit the race condition in the unit test where we saw the crash.

Reply:
Without additional information from the Davix internals, if the use of `dptr`/`dptr->davixPosix` needs a lock here, I have to conclude that it needs it every where :(